### PR TITLE
Feat(Enhancement): Landing Zoom

### PR DESCRIPTION
**Pull request by Koranir**
## Feature Details
Adds a zoom-in when landing (+ a preference in "other").

## UI Screenshots

https://user-images.githubusercontent.com/101683475/230716611-9a996dc3-3df2-4604-8f45-949a1f52fa68.mp4

### CODE REVIEW

File: source/Engine.h [hunk 1]
```diff
@@ -252,8 +252,10 @@ class Engine {
 	TestContext *testContext = nullptr;
 
 	double zoom = 1.;
+	double trueZoom = 1.;
```
Comment (RisingLeaf) [inline source/Engine.h:255]: trueZoom is not exactly precise naming as the zoom the player experiences is another

File: source/Engine.cpp [hunk 2]
```diff
@@ -552,28 +552,29 @@ void Engine::Step(bool isActive)
...
 			// Define zoom speed bounds to prevent asymptotic behavior.
 			static const double MAX_SPEED = .05;
 			static const double MIN_SPEED = .002;
 
-			double zoomRatio = max(MIN_SPEED, min(MAX_SPEED, abs(log2(zoom) - log2(zoomTarget)) * ZOOM_SPEED));
-			if(zoom < zoomTarget)
-				nextZoom = min(zoomTarget, zoom * (1. + zoomRatio));
-			else if(zoom > zoomTarget)
-				nextZoom = max(zoomTarget, zoom * (1. / (1. + zoomRatio)));
+			double zoomRatio = max(MIN_SPEED, min(MAX_SPEED, abs(log2(trueZoom) - log2(zoomTarget)) * ZOOM_SPEED));
```
Comment (RisingLeaf) [inline source/Engine.cpp:570]: line too long
```suggestion
			double zoomRatio = max(MIN_SPEED, min(MAX_SPEED, abs(log2(trueZoom) - log2(zoomTarget))
				* ZOOM_SPEED));
```

**Review by RisingLeaf — CHANGES_REQUESTED** [review]

File: source/Engine.cpp (hunk 2 above)
Comment (Koranir) [reply source/Engine.cpp:570]: How long is too long?

File: source/Engine.h (hunk 1 above)
Comment (Koranir) [reply source/Engine.h]: renamed to baseZoom

File: source/Engine.cpp (hunk 2 above)
Comment (RisingLeaf) [reply source/Engine.cpp]: maximum 100

File: source/Engine.h (hunk 1 above)
Comment (RisingLeaf) [reply source/Engine.h]: thank you

File: source/Engine.cpp (hunk 2 above)
Comment (Koranir) [reply source/Engine.cpp]: > Try to keep lines to about 80 characters in length, and wrap anything over 120 characters.
The style guide says that wrapping is necessary over 120. Wrapping it now would damage the flow of the code.

**Review by Hecter94 — APPROVED** [review]
I can't speak to the code, but I like this conceptually and think it looks great. The smooth zoom-in to the planetary interface UI looks very professional and clean. 
This is an improvement.

**Comment by bene-dictator** [comment]
I don't like the idea, but I'd be happy as long as there's an option to toggle it off.

**Review by RisingLeaf — APPROVED** [review]

**Comment by TomGoodIdea** [comment]
"Cinematic effects" preference sounds too generic to me, I'd suggest "Dynamic zoom" or something like that.

**Comment by Hurleveur** [comment]
I think that if someone does smth with the camera some ppl disike it should be a different pref. Being generic is not good

File: source/Engine.cpp [hunk 3]
```diff
@@ -246,7 +246,9 @@ Engine::Engine(PlayerInfo &player)
 	isMouseToggleEnabled(Preferences::Has("alt-mouse turning")),
 	ammoDisplay(player), shipCollisions(256u, 32u)
 {
-	zoom = Preferences::ViewZoom();
+	baseZoom = Preferences::ViewZoom();
+	zoomMod = Preferences::Has("Cinematic effects") ? 2. : 1.;
```
Comment (Amazinite) [inline source/Engine.cpp:250]: Looks like you're still using the old preference name here.

Comment (Koranir) [reply source/Engine.cpp]: Thanks for the catch-on

**Review by Saugia — APPROVED** [review]

File: source/Engine.h [hunk 4]
```diff
@@ -261,8 +261,10 @@ class Engine {
 	TestContext *testContext = nullptr;
 
 	double zoom = 1.;
+	double baseZoom = 1.;
 	// Tracks the next zoom change so that objects aren't drawn at different zooms in a single frame.
 	double nextZoom = 0.;
+	double zoomMod = 2.;
```
Comment (Amazinite) [inline source/Engine.h:267]: I know this gets set before it's used, but it feels weird to me for the "default" to be 2.
```suggestion
	double zoomMod = 1.;
```

**Pull request merged**